From bf8cbea392ba73b09476d8f4fac27802eef37846 Mon Sep 17 00:00:00 2001
From: sukanka <su975853527@gmail.com>
Date: Wed, 4 Jun 2025 20:19:19 +0800
Subject: [PATCH 4/4] system mbedtls

we use system mbedtls here, it works.

4	0	CMakeLists.txt
0	7	externals/CMakeLists.txt
1	1	src/yuzu/CMakeLists.txt
1	1	src/yuzu_cmd/CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 957ef7b907..669965695b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -321,6 +321,10 @@ find_package(stb MODULE)
 find_package(VulkanMemoryAllocator CONFIG)
 find_package(ZLIB 1.2 REQUIRED)
 find_package(zstd 1.5 REQUIRED)
+set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/lib/mbedtls2/pkgconfig")
+pkg_check_modules(MbedTLS REQUIRED IMPORTED_TARGET GLOBAL mbedtls mbedcrypto)
+include_directories(${MbedTLS_INCLUDE_DIRS})
+link_directories(${MbedTLS_LIBRARY_DIRS})
 
 if (NOT YUZU_USE_EXTERNAL_VULKAN_HEADERS)
     find_package(VulkanHeaders 1.3.274 REQUIRED)
diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 754800ed16..bdc1e8e163 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -40,13 +40,6 @@ endif()
 add_subdirectory(glad)
 
 # mbedtls
-add_subdirectory(mbedtls)
-target_include_directories(mbedtls PUBLIC ./mbedtls/include)
-if (NOT MSVC)
-    target_compile_options(mbedcrypto PRIVATE
-        -Wno-unused-but-set-variable
-        -Wno-string-concatenation)
-endif()
 
 # MicroProfile
 add_library(microprofile INTERFACE)
diff --git a/src/yuzu/CMakeLists.txt b/src/yuzu/CMakeLists.txt
index 440ef3c46d..0562c9a818 100644
--- a/src/yuzu/CMakeLists.txt
+++ b/src/yuzu/CMakeLists.txt
@@ -397,7 +397,7 @@ endif()
 target_link_libraries(yuzu PRIVATE common core input_common frontend_common network video_core)
 target_link_libraries(yuzu PRIVATE Boost::headers glad Qt6::Widgets)
 target_link_libraries(yuzu PRIVATE ${PLATFORM_LIBRARIES} Threads::Threads)
-
+target_link_libraries(yuzu PRIVATE mbedtls mbedcrypto)
 target_link_libraries(yuzu PRIVATE Vulkan::Headers)
 if (NOT WIN32)
     target_include_directories(yuzu PRIVATE ${Qt6Gui_PRIVATE_INCLUDE_DIRS})
diff --git a/src/yuzu_cmd/CMakeLists.txt b/src/yuzu_cmd/CMakeLists.txt
index ebd8fd7387..20c2c285d9 100644
--- a/src/yuzu_cmd/CMakeLists.txt
+++ b/src/yuzu_cmd/CMakeLists.txt
@@ -29,7 +29,7 @@ add_executable(yuzu-cmd
 )
 
 target_link_libraries(yuzu-cmd PRIVATE common core input_common frontend_common)
-target_link_libraries(yuzu-cmd PRIVATE glad)
+target_link_libraries(yuzu-cmd PRIVATE glad mbedtls mbedcrypto)
 if (MSVC)
     target_link_libraries(yuzu-cmd PRIVATE getopt)
 endif()
-- 
2.49.0

